src/parser_private.hh

```diff
@@ -76,6 +76,16 @@ namespace sdf
       const std::string &_source, bool _convert, const ParserConfig &_config,
       Errors &_errors);
 
+  /// \brief Perform a series of checks to determine the validity of this XML
+  /// document in the context of an SDF Element. This is called before
+  /// readXml.
+  /// \remark For internal use only. Do not use this function.
+  /// \param[in] _xml Pointer to the TinyXML element.
+  /// \param[in] _source Source of the XML document.
+  /// \param[in] _errors Captures errors found during the checks.
+  bool checkXml(tinyxml2::XMLElement *_xml, const std::string &_source,
```

Comment:
Can you mention in the documentation that the `_xml` must point to the root level element? I would also suggest changing the name of the function to make that more clear. Otherwise, someone might be tempted to use it just like `readXml`, which works on all levels of the xml hierarchy, not just the top.

Reply:
Makes sense, let me know what you think, thanks! https://github.com/ignitionrobotics/sdformat/pull/636/commits/87d3c91873328221a1deeed46f877881d2ce765d

Reply:
That's perfect! Thanks.